app/components/channel_drawer/channels_list/channel_item/channel_item.js

```diff
@@ -93,25 +127,28 @@ export default class ChannelItem extends PureComponent {
         );
 
         return (
-            <TouchableHighlight
-                underlayColor={changeOpacity(theme.sidebarTextHoverBg, 0.5)}
-                onPress={this.onPress}
-            >
-                <View style={style.container}>
-                    {extraBorder}
-                    <View style={[style.item, extraItemStyle]}>
-                        {icon}
-                        <Text
-                            style={[style.text, extraTextStyle]}
-                            ellipsizeMode='tail'
-                            numberOfLines={1}
-                        >
-                            {displayName}
-                        </Text>
-                        {badge}
+            <AnimatedView ref={this.setPreviewRef}>
```

Comment:
and then `(val) => { this.previewRef = val; }`

Reply:
we should be avoiding arrow functions in the props of the components as they create a new function each time the component renders ;)

Reply:
We're specifically trying to avoid adding an anonymous function here since it causes a new function to be created on each render call which can lead to child components re-rendering. I think React handles it better with refs, but we generally try to avoid either binding functions or using arrow functions inside of render